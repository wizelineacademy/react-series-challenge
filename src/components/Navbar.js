import React, { Component } from "react";
import { NavLink, Link } from "react-router-dom";
import {UlStyled, LiStyled, InputTextStyled,ButtonTextStyled} from "./Navbar.style"

class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tag: "nothing"
    };
  }
  render() {
    return (
      <React.Fragment>
        <UlStyled>
          <LiStyled>
            <NavLink to="/">Home</NavLink>
          </LiStyled>
          <LiStyled>
          <NavLink to="/fav">Favs</NavLink>
          </LiStyled>
          <LiStyled>
          <InputTextStyled
              type="text"
              name="search"
              onChange={event => { this.setState({ tag: event.target.value }); }}
            />
          </LiStyled>
          <LiStyled>
            <Link
              to={{
                pathname: "/search",
                state: { query: this.state.tag }
              }}
            >
              <ButtonTextStyled type="button" id="add">Search</ButtonTextStyled>
            </Link>
          </LiStyled>
        </UlStyled>
      </React.Fragment>
    );
  }
}

export default Navbar;