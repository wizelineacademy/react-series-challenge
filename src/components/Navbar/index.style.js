import styled from 'styled-components';
import { NavLink as RouterLink } from 'react-router-dom';

export const Nav = styled.ul`
  position: absolute;
  top: 10px;
  left: 10px;
  right: 10px;
  height: 60px;
  margin: 0;
  padding: 0;
  list-style: none;
  background: #222;
  font-family: 'Oswald', sans-serif;
  font-size: 18px;
  font-weight: 300;
`;

export const NavLi = styled.li`
  position: relative;
  float: left;
  line-height: 60px;
  background: inherit;
  text-align: center;
  transition: all 0.2s;

  :before {
    content: '';
    display: block;
    position: absolute;
    left: 50%;
    margin-left: -30px;
    width: 60px;
    height: 60px;
    background: #222;
    border-radius: 50%;
    transform: rotate(45deg);
    transition: all 0, background 0.2s;
  }

  :hover:before {
    margin-top: 1px;
    border-radius: 50% 50% 0 50%;
    transition: all 0.1s, background 0.2s,
      margin-top 0.2s cubic-bezier(0.5, 30, 0.2, 0);
  }

  :hover,
  :hover:before {
    background: #3a3a3a;
  }
`;

export const ReactRouterNavLink = styled(RouterLink)`
  position: relative;
  display: block;
  padding: 0 20px;
  line-height: inherit;
  color: white;
  text-decoration: none;
`;
