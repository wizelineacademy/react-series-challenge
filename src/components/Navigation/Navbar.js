import React from "react";
import { withRouter } from "react-router-dom"
import SearchBar from "../SearchBar/SearchBar";
import { ContainerNavbarStyled, NavbarLinkStyle } from "./NavbarStyle";

const Navbar = ({ location }) => (
  // location.pathame
  <ContainerNavbarStyled>
    <div>
      <NavbarLinkStyle exact to="/">Home</NavbarLinkStyle>
      <NavbarLinkStyle to="/favorites">Favoritos</NavbarLinkStyle>
    </div>
    <SearchBar from={location.pathname}/>
  </ContainerNavbarStyled>
);

export default withRouter(Navbar);
