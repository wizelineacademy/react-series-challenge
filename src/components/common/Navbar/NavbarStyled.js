
import styled from 'styled-components'
import Theme from '../Theme'



const { primary } = Theme

const NavbarStyled = styled.nav`
color: #fff;
background-color: ${primary};
width: 100%;
height: 56px;
line-height: 56px;
display:flex;
justify-content:center;
align-items:center;

margin:0;
`


export default NavbarStyled