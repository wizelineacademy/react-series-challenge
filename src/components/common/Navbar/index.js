import React from 'react'
import NavbarStyled from './NavbarStyled'
import Link from '../Link'



const Navbar = (props)=>{
    return(
        <NavbarStyled>
                <Link to="/">Home</Link>
                <Link to="/favorites">Favorites</Link>
        </NavbarStyled>
    )
}

export default Navbar