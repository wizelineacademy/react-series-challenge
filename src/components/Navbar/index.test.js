import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './index';

describe('Navbar Component', () => {
  it('should render the navbar correctly', () => {
    const component = shallow(<Navbar />);
    expect(component).toMatchSnapshot();
  });
});
