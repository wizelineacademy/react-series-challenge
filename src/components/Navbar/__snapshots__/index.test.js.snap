// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render the navbar correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.li>
          <Styled(NavLink)
                    to="/"
          >
                    Home
          </Styled(NavLink)>
</styled.li>,
        <styled.li>
          <Styled(NavLink)
                    to="/favorites"
          >
                    Favorites
          </Styled(NavLink)>
</styled.li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(NavLink)
            to="/"
>
            Home
</Styled(NavLink)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(NavLink)
            to="/favorites"
>
            Favorites
</Styled(NavLink)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Favorites",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.li>
            <Styled(NavLink)
                        to="/"
            >
                        Home
            </Styled(NavLink)>
</styled.li>,
          <styled.li>
            <Styled(NavLink)
                        to="/favorites"
            >
                        Favorites
            </Styled(NavLink)>
</styled.li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(NavLink)
              to="/"
>
              Home
</Styled(NavLink)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(NavLink)
              to="/favorites"
>
              Favorites
</Styled(NavLink)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Favorites",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
